File: be-fschool/src/main/java/com/tinder/befschool/entity/Schedule.java
package com.tinder.befschool.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;

@Entity
@Table(
        name = "schedules",
        indexes = {
                @Index(
                        name = "idx_schedule_class_slot",
                        columnList = "class_id,school_year,semester,day_of_week,period"
                ),
                @Index(
                        name = "idx_schedule_teacher_slot",
                        columnList = "teacher_id,school_year,semester,day_of_week,period"
                )
        }
)
public class Schedule extends Auditable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "class_id")
    private Long classId;

    @Column(name = "class_name")
    private String className;

    @NotNull
    @Column(name = "day_of_week", nullable = false)
    private Integer dayOfWeek;

    @Column(name = "period")
    private String period;

    @Column(name = "subject_id")
    private Long subjectId;

    @Column(name = "subject")
    private String subject;

    @Column(name = "teacher")
    private String teacher;

    @Column(name = "teacher_id")
    private Long teacherId;

    @Column(name = "room")
    private String room;

    @Column(name = "school_year")
    private String schoolYear;

    @Column(name = "semester")
    private Integer semester;

    @Column(name = "start_time")
    private String startTime;

    @Column(name = "end_time")
    private String endTime;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getClassId() {
        return classId;
    }

    public void setClassId(Long classId) {
        this.classId = classId;
    }

    public String getClassName() {
        return className;
    }

    public void setClassName(String className) {
        this.className = className;
    }

    public Integer getDayOfWeek() {
        return dayOfWeek;
    }

    public void setDayOfWeek(Integer dayOfWeek) {
        this.dayOfWeek = dayOfWeek;
    }

    public String getPeriod() {
        return period;
    }

    public void setPeriod(String period) {
        this.period = period;
    }

    public Long getSubjectId() {
        return subjectId;
    }

    public void setSubjectId(Long subjectId) {
        this.subjectId = subjectId;
    }

    public String getSubject() {
        return subject;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

    public String getTeacher() {
        return teacher;
    }

    public void setTeacher(String teacher) {
        this.teacher = teacher;
    }

    public Long getTeacherId() {
        return teacherId;
    }

    public void setTeacherId(Long teacherId) {
        this.teacherId = teacherId;
    }

    public String getRoom() {
        return room;
    }

    public void setRoom(String room) {
        this.room = room;
    }

    public String getSchoolYear() {
        return schoolYear;
    }

    public void setSchoolYear(String schoolYear) {
        this.schoolYear = schoolYear;
    }

    public Integer getSemester() {
        return semester;
    }

    public void setSemester(Integer semester) {
        this.semester = semester;
    }

    public String getStartTime() {
        return startTime;
    }

    public void setStartTime(String startTime) {
        this.startTime = startTime;
    }

    public String getEndTime() {
        return endTime;
    }

    public void setEndTime(String endTime) {
        this.endTime = endTime;
    }
}

File: prm/lib/models/schedule.dart
class Schedule {
  final int id;
  final String className;
  final int dayOfWeek;
  final String period;
  final String subject;
  final String teacher;
  final int? teacherId;
  final String room;
  final String startTime;
  final String endTime;

  Schedule({
    required this.id,
    required this.className,
    required this.dayOfWeek,
    required this.period,
    required this.subject,
    required this.teacher,
    this.teacherId,
    required this.room,
    required this.startTime,
    required this.endTime,
  });

  factory Schedule.fromJson(Map<String, dynamic> json) {
    return Schedule(
      id: json['id'],
      className: json['className'],
      dayOfWeek: json['dayOfWeek'],
      period: json['period'],
      subject: json['subject'],
      teacher: json['teacher'] ?? '',
      teacherId: json['teacherId'] as int?,
      room: json['room'] ?? '',
      startTime: json['startTime'] ?? '',
      endTime: json['endTime'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'className': className,
      'dayOfWeek': dayOfWeek,
      'period': period,
      'subject': subject,
      'teacher': teacher,
      'teacherId': teacherId,
      'room': room,
      'startTime': startTime,
      'endTime': endTime,
    };
  }
}
